base/linalg/lapack.jl

```diff
@@ -441,6 +486,91 @@ for (gebrd, gelqf, geqlf, geqrf, geqp3, geqrt, geqrt3, gerqf, getrf, elty, relty
     end
 end
 
+"""
+    gebrd!(A) -> (A, d, e, tauq, taup)
+
+Reduce `A` in-place to bidiagonal form `A = QBP'`. Returns `A`, containing the
+bidiagonal matrix `B`, `d`, containing the diagonal elements of `B`, `e`,
+containing the off-diagonal elements of `B`, `taup`, containing the
+elementary reflectors representing `P`, and `tauq` containing the
+elementary reflectors representing `Q`.
+"""
+gebrd!(A::StridedMatrix)
+
+"""
+    gelqf!(A, tau)
+
+Compute the `LQ` factorization of `A`, `A = LQ`. `tau` contains scalars
+which parameterize the elementary reflectors of the factorization. Returns
+`A` and `tau` modified in-place.
+"""
+gelqf!(A::StridedMatrix, tau::Vector)
+
+"""
+    geqlf!(A, tau)
+
+Compute the `QL` factorization of `A`, `A = QL`. `tau` contains scalars
+which parameterize the elementary reflectors of the factorization. Returns
+`A` and `tau` modified in-place.
+"""
+geqlf!(A::StridedMatrix, tau::Vector)
+
+"""
+    geqp3!(A, jpvt, tau)
+
+Compute the pivoted `QR` factorization of `A`, `AP = QR` using BLAS level 3.
+`P` is a pivoting matrix, represented by `jpvt`. `tau` stores the elementary
+reflectors. `A`, `jpvt`, and `tau` are modified in-place.
+"""
+geqp3!(A::StridedMatrix, jpvt::Vector{BlasInt}, tau::Vector)
+
+"""
+    geqrt!(A, T)
```

Comment:
what dimensions does T need to have? and does this literally return `-> (A, T)` ?